src/shell/shell_cmds.cpp

```diff
@@ -740,6 +741,74 @@ void DOS_Shell::CMD_DIR(char * args) {
 	dos.dta(save_dta);
 }
 
+void DOS_Shell::CMD_LS(char *args)
```

Comment:
I'm not sure how direct `args` is (if it's processed further up the API chain), but might want to check for null. 
Further down in the error-scenario `trim(args)` could dereference the `nullptr`.
https://onlinegdb.com/HyDgmTHjL

Reply:
Good point. I think we dealt/resolved with this thanks to Coverity flagging this issue, but I don't remember the specifics any more - I will investigate.

Reply:
Good news: I am ~99% sure there is no way to forcefully insert null into this command; going up the call stack for internal commands we end up with locally-allocated buffer (`src/shell/shell.cpp:329`), which is then passed down through several functions, each one trimming down the input and doing some processing on the line.

That being said, this solution is not great… uncareful refactorization might easily introduce an error here. *Some* mechanism to protect the code here would be in order, but the more I look at it, the more it turns into a serious refactorization effort, going way beyond scope of a this new command.

I would say: to be considered in a follow-up PR. Perhaps by adding GCC attribute `nonnull` with -Wnonnull (this would allow compiler to catch obvious cases and static analysis to catch more advanced ones, if someone will introduce problematic code in the future).

Reply:
> Some mechanism to protect the code here would be in order, but the more I look at it, the more it turns into a serious refactorization effort, going way beyond scope of a this new command.

Agree that this is best kept to a stand-alone PR; sounds like that work would justify a PR in itself.